packages/sanity/src/_internal/cli/commands/backup/downloadBackupCommand.ts

```diff
@@ -0,0 +1,297 @@
+  --overwrite          Allows overwriting of existing backup file.
+  --concurrency <num>  Concurrent number of backup item downloads. (max: 24)
+
+Examples
+  sanity backup download DATASET_NAME --backup-id 2024-01-01-backup-1
+  sanity backup download DATASET_NAME --backup-id 2024-01-01-backup-2 --out /path/to/file
+  sanity backup download DATASET_NAME --backup-id 2024-01-01-backup-3 --out /path/to/file --overwrite
+`
+
+const downloadBackupCommand: CliCommandDefinition = {
+  name: 'download',
+  group: 'backup',
+  signature: '[DATASET_NAME]',
+  description: 'Download a dataset backup to a local file.',
+  helpText,
+  // eslint-disable-next-line max-statements
+  action: async (args, context) => {
+    const {output, chalk} = context
+    const [client, opts] = await prepareBackupOptions(context, args)
+    const {projectId, datasetName, backupId, outDir, outFileName} = opts
+
+    // If any of the output path or file name is empty, cancel the operation.
+    if (outDir === '' || outFileName === '') {
+      output.print('Operation cancelled.')
+      return
+    }
+    const outFilePath = path.join(outDir, outFileName)
+
+    output.print('╭───────────────────────────────────────────────────────────╮')
+    output.print('│                                                           │')
+    output.print('│ Downloading backup for:                                   │')
+    output.print(`│ ${chalk.bold('projectId')}: ${chalk.cyan(projectId).padEnd(56)} │`)
+    output.print(`│ ${chalk.bold('dataset')}: ${chalk.cyan(datasetName).padEnd(58)} │`)
+    output.print(`│ ${chalk.bold('backupId')}: ${chalk.cyan(backupId).padEnd(56)} │`)
+    output.print('│                                                           │')
+    output.print('╰───────────────────────────────────────────────────────────╯')
+    output.print('')
+    output.print(`Downloading backup to "${chalk.cyan(outFilePath)}"`)
+
+    const start = Date.now()
+    const progressSpinner = newProgress(output, 'Setting up backup environment...')
+
+    // Create temporary directory to store files before bundling them into the archive at outputPath.
+    // We are adding unix milliseconds and a random key to try to back up in a unique location on each attempt.
+    // Temporary directories are normally deleted at the end of backup process, any unexpected exit may leave them
+    // behind, hence it is important to create a unique directory for each attempt.
+    // We intentionally avoid `datasetName` and `backupId`in path since some operating system have a max length
+    // of 256 chars on path names and both of them can be quite long in some cases.
+    const tmpOutDir = path.join(tmpdir(), `backup-${Date.now()}-${randomKey(5)}`)
+
+    // Create required directories if they don't exist.
+    for (const dir of [
+      outDir,
+      tmpOutDir,
+      path.join(tmpOutDir, 'images'),
+      path.join(tmpOutDir, 'files'),
+    ]) {
+      if (!existsSync(dir)) {
```

Comment:
You can remove this check as it's safe to call `mkdirSync` even if the directory exists when `recursive: true`. This can happen even _with_ the check as the directory could be created by another process after checking that it doesn't exist.

Reply:
Good point, removed.